src/main/steps/c100-rebuild/child-details/add-children/postController.ts

```diff
@@ -0,0 +1,197 @@
+        this.addAnotherChild(req);
+        this.resetSessionTemporaryFormValues(req);
+        req.session.userCase.tempchildrenFormData = undefined;
+        const redirectURI =
+          C100_children_DETAILS_PARENTIAL_RESPONSIBILITY + `?childId=${req.session.userCase?.children?.[0].id}`;
+        return super.redirect(req, res, redirectURI);
+      } else {
+        this.mapEnteriesToValuesAfterContinuing(req, res);
+      }
+    } else {
+      const fields = typeof this.fields === 'function' ? this.fields(req.session.userCase) : this.fields;
+      const form = new Form(fields);
+      const { _csrf, ...formData } = form.getParsedBody(req.body);
+      this.errorsAndRedirect(req, res, formData, form);
+      const { addAnotherChild } = req['body'];
+      switch (addAnotherChild) {
+        case 'Yes':
+          this.addAnotherChild(req);
+          this.resetSessionTemporaryFormValues(req);
+          break;
+        default:
+      }
+      return super.redirect(req, res, C100_children_DETAILS_ADD);
+    }
+  }
+
+  /**
+   * It takes the form data, the form, and the request and response objects, and if there are errors, it
+   * sets the errors in the session and redirects
+   * @param req - AppRequest<AnyObject> - The request object
+   * @param {Response} res - Response - The response object from express
+   * @param {AnyObject} formData - The data that was submitted by the user.
+   * @param {Form} form - The form object that was created in the controller.
+   * @returns The errors from the form.
+   */
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  protected errorsAndRedirect(req: AppRequest<AnyObject>, res: Response, formData: Partial<Case>, form: Form) {
+    req.session.errors = form.getErrors(formData);
+    if (req.session.errors.length) {
+      return super.redirect(req, res, C100_children_DETAILS_ADD);
+    }
+  }
+
+  /**
+   * It takes a request and a response, and returns a redirect to the root path
+   * @param req - AppRequest<AnyObject>
+   * @param {Response} res - Response - this is the response object that will be sent back to the client.
+   * @returns The response body is being returned.
+   */
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  public addAnotherChild(req: AppRequest<AnyObject>): void {
+    const { firstname, lastname } = req['body'];
+    const applicantInformation = {
+      id: uuidv4() as string,
+      firstname,
+      lastname,
+      personalDetails: {
+        DateoBirth: '',
+        isDateOfBirthKnown: '',
+        ApproximateDateOfBirth: '',
+        Sex: '',
+      },
+      childMatter: {
+        isDecisionTaken: '',
+      },
+      parentialResponsibility: {
+        statement: '',
+      },
+    };
+    let applicantInSession: childrenDetails[] = [];
+    if (req.session.userCase.hasOwnProperty('children') && req.session.userCase.children) {
+      applicantInSession = req.session.userCase.children;
```

Comment:
please remove references to applicant since this is specific to children

Reply:
it's been resolved